packages/payment-processor/src/payment/prepare-hinkal.ts

```diff
@@ -0,0 +1,21 @@
+import { Signer } from 'ethers';
+
+import { Hinkal, exportEthersProvider, preProcessing } from '@hinkal/common';
+
+/**
+ * prepares wrapper hinkal object for private transaction
+ * @param signer signer object to be passed
+ */
+export const prepareHinkal = async (signer: Signer): Promise<Hinkal<unknown>> => {
+  await preProcessing();
+  const hinkal = new Hinkal();
+
```

Comment:
We decided not to add a try / catch block, since it would just re-throw an already existing error with a slightly different error message we do not see much benefit in adding this. In addition it would make it harder to detect the exact type of error thrown.